### DISCUSSION TITLE
View Colors

### DISCUSSION THREAD

**Discussion by StefanGreve** [tui-cs/Terminal.Gui#1667] [Q&A] [answered]
According to the [conceptual documentation](https://migueldeicaza.github.io/gui.cs/articles/overview.html#colors-and-color-schemes),

---

*Some views support setting individual color attributes, you create an attribute for a particular pair of Foreground/Background like this:*

```csharp
var myColor = Application.Driver.MakeAttribute (Color.Blue, Color.Red);
var label = new Label (...);
label.TextColor = myColor
```

---

However, this text segment did not elaborate any further under *which* conditions some views support this type of color assignment. For Instance, a label I created in my Views such as

```csharp
Label titleLabel = new(ResourceManager.GetString("LoginTitleLabel")) { X = 1, Y = 1 };
```

does not have a `TextColor` attribute. I also found this issue #1097 which discusses the possibility of adding attributed strings to the library, are there any plans to move forward with this idea? While I love the idea of defining color schemes as prescribed in the current version because it encourages consistent design practices, sometimes it's a good idea to highlight an important piece of information in a string with color based on custom rules. Being able to change the color of individual text labels would be pretty cool.

Of course, perhaps there already exists a way to add colors to specific text labels that I have missed during my research? Far be it from me to rule out this possibility! Anyway, that's my only pet peeve about this library, love everything else about it ❤️

**Comment by BDisp** [comment, marked as answer]
There is no TextColor property anymore. That documentation must be updated. It was superseded by the `ColorScheme` property. Use it like this on a predefined:
`label.ColorScheme = Colors.Base.Normal`
or create a new `ColorScheme` with the `Toplevel`, `Base`, `Menu`, `Dialog `or `Error`.

About the use of escape sequence for the colors, you can take a look how the `NetDriver` handles it.

**Reply by StefanGreve** [reply]
Hm right that would be possible. But why was the `TextColor` property deprecated? In my opinion, it would be much easier to have something like `label.TextColor = (Color.Green, Color.Black)` to overwrite existing color schemes. I already use color schemes and generally I think they're great, but in some cases I want to use a different color for a view that would otherwise be using the color as defined in `Colors.Base.Normal` if you understand what I mean?

On an unrelated note, could it also possibly be the case that the current README also contains outdated code examples? For example for a Button key binding, the methods here

```csharp
var btn = new Button ("Press Me");
btn.ClearKeybinding (Command.Accept);
btn.AddKeyBinding (Key.b, Command.Accept);
```

don't seem to be available anymore.

**Reply by BDisp** [reply]
On the contrary, the `Keybinding` is a recent feature. The `TextColor` was deprecated before I contribute for this repo, but you can have any number of `ColorScheme` and this allow to have `Normal`, `Focus`, `HotNormal,` `HotFocus`, `Disabled` for each the categories I did mentioned above.. You can handle in the way you want.

**Reply by StefanGreve** [reply]
Oh why of course, it somehow didn't get into my head that there should be no limit to the number of color schemes one could have. Thanks for clarifying that!
